src/caformat-util.c

```diff
@@ -156,7 +156,7 @@ int ca_with_feature_flags_format(uint64_t features, char **ret) {
                 features &= ~f;
         }
 
-        if ((features & ~(CA_FORMAT_EXCLUDE_NODUMP|CA_FORMAT_EXCLUDE_SUBMOUNTS|CA_FORMAT_SHA512_256)) != 0) {
+        if ((features & ~(CA_FORMAT_EXCLUDE_NODUMP|CA_FORMAT_EXCLUDE_SUBMOUNTS|CA_FORMAT_EXCLUDE_FILE|CA_FORMAT_SHA512_256)) != 0) {
```

Comment:
I don't like that we are mixing two things: how we are generating a list of files to serialize, with what information we are serializing about this list of files. The result should not depend on the presence of `--exclude-file=no` on the command line, but only on the presence of files that would be excluded. I.e. if there are no files that would be excluded by this, or not submounts, or no files with nodumps flags, the result should be identical no matter if any of the `--exclude…` flags are used.

I have to go now, so I'm leaving just this short note for now, I'll expand on this later.